spec/takeaway_spec.rb

```diff
@@ -0,0 +1,73 @@
+require "takeaway"
+
+describe Takeaway do
+
+  TEST_MENU = { "Won Ton" => 3.70,
+                "Chicken and Noodle Soup" => 1.90,
+                "Pork Chow Mein" => 4.60,
+              }
+
+  subject(:takeaway){described_class.new(menu: TEST_MENU, notification_class: notification_class)}
+  let(:notification_class) {double(:notification_class, new: notification)}
+  let(:notification){double(:notification)}
+
+  describe "before order" do
+
+    it "returns an item and its price" do
+      expect(takeaway.menu).not_to be_empty
+    end
+
+    it "returns empty basket message" do
+      expect(takeaway.basket_summary).to eq "Your basket is empty"
+    end
+  end
+
+  describe "adds menu items to basket" do
+
+    it "adds one item by default" do
+      expect(takeaway.order("Won Ton")).to eq "1x Won Ton added to your basket."
+    end
+
+    it "adds an item multiple times" do
+      expect(takeaway.order("Won Ton", 2)).to eq "2x Won Ton added to your basket."
+    end
+
+    context "orders multiple items" do
+
+      before do
+        takeaway.order("Won Ton", 3)
+        takeaway.order("Pork Chow Mein")
+      end
+
+      it "returns a total" do
+        expect(takeaway.total).to eq "Total: 15.7" #to do - handle currency correctly
+      end
+
+      it "returns a basket summary" do
+        expect(takeaway.basket_summary).to eq "Won Ton x 3 = 11.1,Pork Chow Mein x 1 = 4.6"
```

Comment:
Metrics/LineLength: Line is too long. [91/80] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)